tests/integration/harvest/test_compare.py

```diff
@@ -36,3 +38,50 @@ def test_compare(
         assert len(harvest_source.compare_data["delete"]) == 1
 
     # TODO: add sort test
+
+    def test_write_compare_to_db(
+        self,
+        organization_data,
+        source_data_dcatus,
+        job_data_dcatus,
+        interface,
+        internal_compare_data,
+    ):
+        # add the necessary records to satisfy FKs
+        interface.add_organization(organization_data)
+        interface.add_harvest_source(source_data_dcatus)
+        interface.add_harvest_job(job_data_dcatus)
+
+        # prefill with records
+        records = []
+        for record in internal_compare_data["records"]:
+            records.append(
+                {
+                    "identifier": record["identifier"],
+                    "harvest_job_id": job_data_dcatus["id"],
+                    "harvest_source_id": job_data_dcatus["harvest_source_id"],
+                    "source_hash": dataset_to_hash(sort_dataset(record)),
+                    "source_raw": json.dumps(record),
+                }
+            )
+
+        interface.add_harvest_records(records)
+
+        harvest_source = HarvestSource(job_data_dcatus["id"])
+        harvest_source.get_record_changes()
+
+        harvest_source.write_compare_to_db()
+
+        expected = sorted(
+            [
+                "cftc-dc3",
+                "cftc-dc7",
+                "cftc-dc5",
+                "cftc-dc4",
+                "cftc-dc6",
+                "cftc-dc1",
+                "cftc-dc2",
+            ]
+        )
```

Comment:
It's a little confusing that you used the sort function to create the expected; but of course it works. Just gave me pause reading it.

Reply:
this is an old test. i had to move it b/c it does something to the DB which interferes with the pagination tests above. more than likely it's the `harvest_source.write_compare_to_db()`. leaving it where it was mucks up pagination for some reason. definitely worth considering if this indicates a deeper bug or just a symptom of a weird test scenario though. 

i could also set conftest to clean db after every function instead of just the session but that slows things down dramatically and this seemed the better route.

Reply:
if i left this where it was, this assertion wouldn't pass on first run, but would pass when i ran it isolation
https://github.com/GSA/datagov-harvester/pull/104/files#diff-4f46b3a8f79b223696492f1254bdf0ec30ea2b66c04bb3388e2d0aaacc55d8e0R531